src/engine/polyline_compressor.cpp

```diff
@@ -13,9 +13,8 @@ namespace engine
 namespace /*detail*/ // anonymous to keep TU local
 {
 
-std::string encode(int number_to_encode)
+void encode(int number_to_encode, std::string& output)
```

Comment:
`perf` showed a lot of append operations here. The char-wise append can't be helped but appending the sub-strings just created a lot of unnecessary copies.